packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm

```diff
@@ -34,6 +34,7 @@ @implementation RCTBaseTextInputView {
 
 - (void)reactUpdateResponderOffsetForScrollView:(RCTScrollView *)scrollView
 {
+  scrollView.firstResponderFocusView = self.backedTextInputView;
```

Comment:
This should respect the early return.

Reply:
@javache I optimized the code to only store it when the first responder is outside the scroll view.